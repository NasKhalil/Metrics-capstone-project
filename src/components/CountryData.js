import React from 'react';
import PropTypes from 'prop-types';
import { BsArrowRightCircleFill } from 'react-icons/bs';

import virus from '../images/virus.png';
import '../styling/CountryData.css';

const CountryData = ({ name, confirmedCases }) => (
  <li className="country-data">
    <div className="right">
      <BsArrowRightCircleFill className="arrow" />
    </div>
    <div className="country">
      <img src={virus} alt="Covid logo" className="covid-img" />
      <aside>
        <h2>{name}</h2>
        <span>Total Confirmed</span>
        <h3>{`${confirmedCases} cases`}</h3>
      </aside>
    </div>
  </li>
);

CountryData.propTypes = {
  name: PropTypes.string.isRequired,
  confirmedCases: PropTypes.number.isRequired,
};

export default CountryData;
