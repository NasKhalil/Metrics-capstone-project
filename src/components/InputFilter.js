import React from 'react';
import PropTypes from 'prop-types';

import '../styling/InputFilter.css';

const InputFilter = ({ filter, onChange }) => (
  <div className="search">
    <input
      type="text"
      value={filter}
      onChange={onChange}
      placeholder="Search Country..."
    />
  </div>
);

InputFilter.propTypes = {
  filter: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default InputFilter;
