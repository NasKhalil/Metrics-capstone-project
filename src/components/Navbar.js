import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { IoIosArrowBack, IoMdMic, IoIosSettings } from 'react-icons/io';

const Navbar = ({ year, title }) => (
  <>
    <nav>
      <div>
        <Link to="/" className="back-btn">
          <IoIosArrowBack />
        </Link>
        <span>{year}</span>
      </div>
      <span>{title}</span>
      <div>
        <Link to="/details" className="back-btn">
          <IoMdMic />
        </Link>
        <IoIosSettings />
      </div>
    </nav>
  </>
);

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
  year: PropTypes.string,
};

Navbar.defaultProps = {
  year: '',
};

export default Navbar;
