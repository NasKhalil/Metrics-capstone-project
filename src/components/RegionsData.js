import React from 'react';
import PropTypes from 'prop-types';

import '../styling/RegionsData.css';

const RegionsData = ({ regions, countryName }) => (
  <>
    <div className="region">
      <h3 className="region-title">Region&apos;s State</h3>
      <ul className="regions">
        {regions && regions.map((region) => {
          const {
            name,
            today_confirmed: todayConfirmed,
            today_deaths: todayDeaths,
            today_recovered: todayRecovered,
          } = region;
          return (
            <li key={name} className="region-list">
              <div>
                <h3 className="region-name">{name}</h3>
                <div className="region-detail">
                  <div>
                    <span>Total Confirmed </span>
                    <h3>{`${todayConfirmed} cases`}</h3>
                  </div>
                  <div>
                    <span>Total Deaths</span>
                    <h3>{`${todayDeaths} cases`}</h3>
                  </div>
                  <div>
                    <span>Total Recovered</span>
                    <h3>{`${todayRecovered} cases`}</h3>
                  </div>
                </div>
              </div>
            </li>
          );
        })}
      </ul>

      {(!regions || !regions[0]) && (
        <h3 className="noData">
          We are sorry!
          <br />
          Data will be available as soon as possible for
          {' '}
          {countryName}
        </h3>
      )}
    </div>
  </>
);

RegionsData.propTypes = {
  regions: PropTypes.instanceOf(Array).isRequired,
  countryName: PropTypes.string.isRequired,
};

export default RegionsData;
