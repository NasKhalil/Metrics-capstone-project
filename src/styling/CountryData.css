.country-data {
  list-style: none;
  width: 80%;
  padding: 30px 10px;
  margin: 0 auto;
  line-height: 1.7;
}

.right {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.arrow {
  font-size: 18px;
}

.country {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.covid-img {
  width: 30px;
  opacity: 0.8;
}

.country-data aside {
  letter-spacing: 4px;
}

.country-data aside h2 {
  font-size: 16px;
}

.country-data aside h3 {
  font-size: 12px;
}

@media all and (min-width: 1024px) {
  .country-data {
    padding: 30px 0;
  }

  .country {
    text-align: start;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    margin-top: 15px;
  }

  .covid-img {
    width: 20%;
  }

  .arrow {
    font-size: 21px;
  }

  .country-data aside h2 {
    font-size: 27px;
  }

  .country-data aside h3 {
    font-size: 21px;
  }
}
