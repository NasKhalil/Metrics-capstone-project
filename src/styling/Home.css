.container {
  padding: 0;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  margin: 0;
  background-color: #ec4c8b;
}

.country-card {
  border-radius: 4px;
}

.country-card a {
  color: #fff;
  text-decoration: none;
}

.country-card:nth-child(even) {
  background-color: #e7317a;
}

.country-card:hover {
  transform: scale(0.97);
}

@media all and (min-width: 768px) {
  .container {
    grid-template-columns: repeat(2, 1fr);
  }

  .country-card:nth-child(even) {
    background: none;
  }

  .country-card:nth-child(4n + 2),
  .country-card:nth-child(4n + 3) {
    background-color: #e7317a;
  }
}
