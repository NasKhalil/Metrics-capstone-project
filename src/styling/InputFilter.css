.search {
  padding: 8px;
}

.search input {
  padding: 6px;
  width: 150px;
  border-radius: 12px;
  border: 1px solid;
}

@media all and (min-width: 768px) {
  .search {
    padding: 16px;
  }

  .search input {
    padding: 14px;
    width: 350px;
    border-radius: 12px;
    border: 1px solid;
  }
}
