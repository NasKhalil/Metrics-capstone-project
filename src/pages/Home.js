import { React, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchApiData } from '../redux/metrics/metrics';
import Navbar from '../components/Navbar';
import World from '../components/World';
import InputFilter from '../components/InputFilter';
import CountryData from '../components/CountryData';

import '../styling/Home.css';

const Home = () => {
  const AllData = useSelector((state) => state.metricsReducer);
  const { oneCountry, totalConfirmed } = AllData;
  const dispatch = useDispatch();

  useEffect(() => {
    if (!AllData.length) {
      dispatch(fetchApiData());
    }
  }, []);

  const [filter, setFilter] = useState('');
  const onChange = (e) => {
    setFilter(e.target.value);
  };

  return (
    <div>
      <Navbar year="2022" title="World Covid-12" />
      <World totalConfirmed={totalConfirmed} />
      <InputFilter filter={filter} onChange={onChange} />
      <ul className="container">
        {oneCountry && oneCountry
          .filter((location) => location.name.toLowerCase().includes(filter.toLowerCase()) || filter === '')
          .map((country) => {
            const {
              name,
              confirmedCases,
              id,
              allDeaths,
              recovered,
              openCases,
              allRegions,
            } = country;
            return (
              <section className="country-card" key={id}>
                <Link
                  to={`/details/${name}`}
                  state={
                    {
                      countryName: name,
                      confirmCases: confirmedCases,
                      deaths: allDeaths,
                      allRecovered: recovered,
                      casesOpen: openCases,
                      regions: allRegions,
                    }
                  }
                >
                  <CountryData name={name} confirmedCases={confirmedCases} />
                </Link>
              </section>
            );
          })}
      </ul>
    </div>
  );
};

export default Home;
