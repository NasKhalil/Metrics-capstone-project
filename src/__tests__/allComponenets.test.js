import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../pages/Home';
import CountryData from '../components/CountryData';
import Navbar from '../components/Navbar';
import InputFilter from '../components/InputFilter';
import RegionsData from '../components/RegionsData';

describe('All components/pages render correctly', () => {
  test('Home component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Home />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CountryData component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <CountryData name="France" confirmedCases={50} />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Navbar component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Navbar title="Test" year="2022" />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('InputFilter component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <InputFilter filter="den" onChange={() => {}} />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('RegionsData component renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <RegionsData regions={[]} countryName="France" />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
