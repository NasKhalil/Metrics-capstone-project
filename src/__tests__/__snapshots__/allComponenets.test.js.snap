// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All components/pages render correctly CountryData component renders correctly 1`] = `
<li
  className="country-data"
>
  <div
    className="right"
  >
    <svg
      className="arrow"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
      />
    </svg>
  </div>
  <div
    className="country"
  >
    <img
      alt="Covid logo"
      className="covid-img"
      src="virus.png"
    />
    <aside>
      <h2>
        France
      </h2>
      <span>
        Total Confirmed
      </span>
      <h3>
        50 cases
      </h3>
    </aside>
  </div>
</li>
`;

exports[`All components/pages render correctly Home component renders correctly 1`] = `
<div>
  <nav>
    <div>
      <a
        className="back-btn"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <span>
        2022
      </span>
    </div>
    <span>
      World Covid-12
    </span>
    <div>
      <a
        className="back-btn"
        href="/details"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 320c37.712 0 68.571-30.924 68.571-68.714V100.714C324.571 62.924 293.712 32 256 32s-68.571 30.924-68.571 68.714v150.572c0 37.79 30.859 68.714 68.571 68.714zm121.139-75.452c0 68.714-58.282 116.815-121.139 116.815s-121.139-48.102-121.139-116.815H96c0 77.873 61.719 143.153 137.144 153.465V480h45.713v-81.987C354.281 386.561 416 322.421 416 244.548h-38.861z"
          />
        </svg>
      </a>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
        />
      </svg>
    </div>
  </nav>
  <div
    className="worldContainer"
  >
    <img
      alt="world map"
      className="worldMap"
      src="world-map.png"
    />
    <div
      className="cases"
    >
      <h3>
        Cases Arround the world
      </h3>
      <span>
        0
      </span>
    </div>
  </div>
  <div
    className="search"
  >
    <input
      onChange={[Function]}
      placeholder="Search Country..."
      type="text"
      value=""
    />
  </div>
  <ul
    className="container"
  />
</div>
`;

exports[`All components/pages render correctly InputFilter component renders correctly 1`] = `
<div
  className="search"
>
  <input
    onChange={[Function]}
    placeholder="Search Country..."
    type="text"
    value="den"
  />
</div>
`;

exports[`All components/pages render correctly Navbar component renders correctly 1`] = `
<nav>
  <div>
    <a
      className="back-btn"
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </a>
    <span>
      2022
    </span>
  </div>
  <span>
    Test
  </span>
  <div>
    <a
      className="back-btn"
      href="/details"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 320c37.712 0 68.571-30.924 68.571-68.714V100.714C324.571 62.924 293.712 32 256 32s-68.571 30.924-68.571 68.714v150.572c0 37.79 30.859 68.714 68.571 68.714zm121.139-75.452c0 68.714-58.282 116.815-121.139 116.815s-121.139-48.102-121.139-116.815H96c0 77.873 61.719 143.153 137.144 153.465V480h45.713v-81.987C354.281 386.561 416 322.421 416 244.548h-38.861z"
        />
      </svg>
    </a>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
      />
    </svg>
  </div>
</nav>
`;

exports[`All components/pages render correctly RegionsData component renders correctly 1`] = `
<div
  className="region"
>
  <h3
    className="region-title"
  >
    Region's State
  </h3>
  <ul
    className="regions"
  />
  <h3
    className="noData"
  >
    We are sorry!
    <br />
    Data will be available as soon as possible for
     
    France
  </h3>
</div>
`;
